3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,218 @@
+  
+
+  //When will we be there??//
+
+  function calculateTravelTime(travelInformation) {
+    const { speed, destinationDistance } = travelInformation;
+    const totalTimeInHours = destinationDistance / speed;
+    const hours = Math.floor(totalTimeInHours);
+    const minutes = Math.round((totalTimeInHours - hours) * 60);
+  
+    const formattedTime = `${hours} hours and ${minutes} minutes`;
+  
+    return formattedTime;
+  }
+  
+  const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+  
+  const travelTime = calculateTravelTime(travelInformation);
+  console.log(travelTime);
+
+
+  //(2) - Series duration of my life// 
+
+  const seriesDurations = [
+    {
+      title: "Money Heist",
+      days: 1,
+      hours: 23,
+      minutes: 50,
+    },
+    {
+      title: "El Chapo",
+      days: 1,
+      hours: 5,
+      minutes: 20,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+  ];
+  
+  function logOutSeriesText() {
+    const totalMinutesInLifespan = 80 * 365 * 24 * 60; // 80 years in minutes
+    let totalMinutesWatchingSeries = 0;
+  
+    for (const series of seriesDurations) {
+      const seriesMinutes = series.days * 24 * 60 + series.hours * 60 + series.minutes;
+      totalMinutesWatchingSeries += seriesMinutes;
+      const percentageOfLife = (seriesMinutes / totalMinutesInLifespan) * 100;
+      console.log(`${series.title} took ${percentageOfLife.toFixed(3)}% of my life`);
+    }
+  
+    const totalPercentage = (totalMinutesWatchingSeries / totalMinutesInLifespan) * 100;
+    console.log(`\nIn total that is ${totalPercentage.toFixed(3)}% of my life`);
+  }
+  
+  logOutSeriesText();
+  
+
+
+
+  //(3) - NOnoN0nOYes (Note taking app)// 3
+
+  const notes = [];
+
+function saveNote(content, id) {
+  
+  if (typeof id !== 'number') {
+    console.log("Error: Invalid ID. Please provide a number.");
+    return;
+  }
+  
+  
+  notes.push({ content, id });
+}
+
+function getNote(id) {
+  
+  if (typeof id !== 'number' || isNaN(id)) {
+    console.log("Error: Invalid ID. Please provide a number.");
+    return;
+  }
+
+  
+  for (const note of notes) {
+    if (note.id === id) {
+      return note;
+    }
+  }
+
+  
+  console.log("Error: Note not found.");
+}
+
+function logOutNotesFormatted() {
+  for (const note of notes) {
+    console.log(`The note with id: ${note.id}, has the following note text: ${note.content}`);
+  }
+}
+
+/
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+logOutNotesFormatted();
+// should log out:
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
```

Comment:
🌟 Love the checks you've implemented in the `saveNote` and `getNote` functions to ensure the provided id is a `number`.
💡 Suggestion: In `getNote`, once you find a note, you could return it immediately without needing to continue the loop. A simple `break` could save some processing time. do some research about it.
❗ It seems there's an unintended slash (`/`) before the `saveNote` function call. This might throw an error.